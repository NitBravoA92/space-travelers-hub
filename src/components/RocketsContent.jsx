import { useSelector } from 'react-redux';
import RocketItem from './RocketItem';

const RocketsContent = () => {
  const { rockets, isLoading, error } = useSelector((state) => state.rockets);

  if (error) {
    return (
      <div className="error-message">
        <h2>{error}</h2>
      </div>
    );
  }

  if (isLoading) {
    return (
      <div className="loading-message">
        <h2>Loading Rockets...</h2>
      </div>
    );
  }

  return (
    <div className="rockets-list">
      { rockets.map((rocket) => (
        <RocketItem key={rocket.id} rocket={rocket} />
      ))}
    </div>
  );
};

export default RocketsContent;
