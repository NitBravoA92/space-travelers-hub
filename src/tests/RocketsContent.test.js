import '@testing-library/jest-dom/extend-expect';
import { screen, fireEvent } from '@testing-library/react';
import RocketsContent from '../components/RocketsContent';
import renderWithProviders from '../helpers/helper-for-test';

describe('The RocketsContent component', () => {
  test('renders correctly into the DOM', () => {
    const { tree } = renderWithProviders(<RocketsContent />);

    expect(tree).toMatchSnapshot();
  });

  test('renders only one rocket item', () => {
    renderWithProviders(<RocketsContent />);

    const renderRocketItems = document.querySelectorAll('.rocket-item');
    expect(renderRocketItems).toHaveLength(1);
  });

  test('renders a reserved rocket', () => {
    renderWithProviders(<RocketsContent />);

    const renderRocketButton = screen.getByRole('button', {
      name: /Cancel Reservation/i,
    });
    expect(renderRocketButton).toBeInTheDocument();
  });

  test("display the button 'Reserve Rocket' when user clicks on the 'Cancel Reservation' Button", async () => {
    const { tree } = renderWithProviders(<RocketsContent />);
    const cancelButton = tree.findByText(/Cancel Reservation/i);

    fireEvent.click(await cancelButton);

    const reserveRocketButton = screen.getByText(/Reserve Rocket/i);
    const buttonClasses = reserveRocketButton.className;

    expect(reserveRocketButton).toBeInTheDocument();
    expect(buttonClasses).toMatch(/btn btn-blue/i);
  });
});
