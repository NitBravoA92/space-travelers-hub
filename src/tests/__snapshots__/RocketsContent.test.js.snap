// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RocketsContent component renders correctly into the DOM 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rockets-list"
      >
        <div
          class="rocket-item"
        >
          <div
            class="rocket-image"
          >
            <img
              alt="Falcon 1"
              src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg"
            />
          </div>
          <div
            class="rocket-details"
          >
            <h2
              class="title"
            >
              Falcon 1
            </h2>
            <p
              class="description"
            >
              <span
                class="badge badge-blue"
              >
                Reserved
              </span>
               
              The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009.
            </p>
            <div
              class="reserve"
            >
              <button
                class="btn btn-gray"
                type="button"
              >
                Cancel Reservation
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rockets-list"
    >
      <div
        class="rocket-item"
      >
        <div
          class="rocket-image"
        >
          <img
            alt="Falcon 1"
            src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg"
          />
        </div>
        <div
          class="rocket-details"
        >
          <h2
            class="title"
          >
            Falcon 1
          </h2>
          <p
            class="description"
          >
            <span
              class="badge badge-blue"
            >
              Reserved
            </span>
             
            The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009.
          </p>
          <div
            class="reserve"
          >
            <button
              class="btn btn-gray"
              type="button"
            >
              Cancel Reservation
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
